source/Halibut.Tests/Support/BackwardsCompatibility/ProxyHalibutTestBinaryRunner.cs

```diff
@@ -128,41 +144,39 @@ async Task ProcessLogs(string s, CancellationToken ct)
 
                         if (s.Contains("RunningAndReady")) hasTentacleStarted.Set();
                     }
+                }, runningTentacleCancellationTokenSource.Token);
 
-                    await Cli.Wrap(new HalibutTestBinaryPath().BinPath(version))
-                        .WithArguments(new string[0])
-                        .WithWorkingDirectory(tmp.FullPath)
-                        .WithStandardOutputPipe(PipeTarget.ToDelegate(ProcessLogs))
-                        .WithStandardErrorPipe(PipeTarget.ToDelegate(ProcessLogs))
-                        .WithEnvironmentVariables(settings)
-                        .ExecuteAsync(cancellationToken);
-                }
-                catch (OperationCanceledException)
+                using var whenAnyCleanupCancellationTokenSource = new CancellationTokenSource();
```

Comment:
For another day, but I'm seeing this type of pattern more and more in this repo. Maybe later we can make some helpful extension methods for this?